pkg/providers/instancetype/types.go

```diff
@@ -173,12 +173,16 @@ func ephemeralStorage(amiFamily amifamily.AMIFamily, blockDeviceMappings []*v1al
 	if len(blockDeviceMappings) != 0 {
 		switch amiFamily.(type) {
 		case *amifamily.Custom:
-			return blockDeviceMappings[len(blockDeviceMappings)-1].EBS.VolumeSize
+			// We can't know if a custom AMI is going to have a volume size.
+			volumeSize := blockDeviceMappings[len(blockDeviceMappings)-1].EBS.VolumeSize
+			if volumeSize != nil {
```

Comment:
This seems reasonable to fallback to the default for the amiFamily. It might be better to look-up the ami default block device mapping so that this is hardcoded as well. We could look into that in a later PR. As a user that specified a customAMI and no volume-size, I'd expect it to use the default block device mapping of the AMI rather than our hardcoded default.

Reply:
Synced offline, going to discover snapshot size in another PR, will open an issue for this

Reply:
https://github.com/aws/karpenter/issues/3934